line_detection_updated.py

```diff
@@ -0,0 +1,302 @@
+'''
+This module creates the functions and class that are used to detect
+straight lines in astronomical images.
+
+Process
+---------
+1. Ingest the raw image
+2. Apply brightness cuts and standardize the image
+3. Perform binary thresholding on the image
+4. Blur the image
+5. Find edges
+6. Apply Hough line on the edge image
+7. Cluster the estimated lines
+8. Return all the estimated lines along with their cluster label
+
+'''
+
+# Importing necessary libraries
+import numpy as np
+import matplotlib.pyplot as plt
+
+import cv2
+
+from skimage.transform import hough_line, hough_line_peaks
+from sklearn.cluster import MeanShift
+
+# Defining functions
+
+
+def image_mask(img, percent):
+    """
+    Provides mask coordinates for an image
+
+    Parameters
+    ----------
+    img : 'numpy.ndarray'
+        The image to be masked
+    percent : `float`
+        percentage of the image borders to be masked out.
+        If x percent is provided, x/2 percent of each edge is masked out
+
+    Returns
+    --------
+    mask coordinates : `int`
+        coordinates of the numpy array at which masking should be applied
+    """
+    x_dim_top = int(img.shape[0]*(percent/2))
+    x_dim_bottom = int(img.shape[0]*(1-percent/2))
+    y_dim_left = int(img.shape[1]*(percent/2))
+    y_dim_right = int(img.shape[1]*(1-percent/2))
+
+    return x_dim_top, x_dim_bottom, y_dim_left, y_dim_right
+
+
+def show(img, ax=None, show=True, title=None, **kwargs):
+    """
+    Show image using matplotlib.
+
+    Parameters
+    ----------
+    img : `np.array`
+        Image to display.
+    ax : `matplotlib.pyplot.Axes` or `None`, optional
+        Ax on which to plot the image. If no axis is given
+        a new figure is created.
+    kwargs : `dict`, optional
+        Keyword arguments that are passed to `imshow`.
+    """
+
+    # make the plots bigger
+    plt.rcParams["figure.figsize"] = (10, 10)
```

Comment:
I would remove these lines from here. If someone gives you an axes on which they want the plot on - they probably want to set a custom size, and if they are not, they are probably using the `rcParams` on top of a notebook to set their own size, or a fine with the default one. 
This would just be overriding those settings.